server/api.ts

```diff
@@ -1104,6 +1105,33 @@ router.postAsync("/stage-stats", ensure.isAdmin, async (req, res) => {
   res.send(updatedStageStats);
 });
 
+/**
+ * POST /api/refetch-stats
+ * Refretches the stats of a batch of mp links in a stage
+ * Params:
+ *   - tourney: identifier for the tourney
+ *   - stage: identifier for the stage
+ * Returns:
+ *   - stats: the updated stage stats
+ */
+router.postAsync("/refetch-stats", ensure.isAdmin, async (req, res) => {
+  logger.info(`${req.user.username} initiated a refresh of ${req.body.tourney}/${req.body.stage} stats`);
+  let mpLinks: string[] = [];
+  if (req.body.stage === "Qualifiers") mpLinks = (await QualifiersLobby.find({ tourney: req.query.tourney })).map(lobby => lobby.link);
+  else mpLinks = (await Match.find({ tourney: req.body.tourney, stage: req.body.stage })).map(match => match.link);
+  for (const mpLink of mpLinks) {
+    if (mpLink) {
+      const found = mpLink.match(mpRegex1) || mpLink.match(mpRegex2);
+      if (found) await fetchMatchAndUpdateStageStats(req.body.tourney, req.body.stage, found[1]);
+    }
+  }
+  const stageStats = await StageStats.findOne({
```

Comment:
can this unnecessary db call be removed by just returning stagestats from `fetchMatchAndUpdateStageStats`?

Reply:
yeah I think I just need to add return statement to that function